src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/web/DocumentController.kt

```diff
@@ -0,0 +1,46 @@
+package org.vaccineimpact.orderlyweb.controllers.web
+
+import org.vaccineimpact.orderlyweb.ActionContext
+import org.vaccineimpact.orderlyweb.FileSystem
+import org.vaccineimpact.orderlyweb.Files
+import org.vaccineimpact.orderlyweb.controllers.Controller
+import org.vaccineimpact.orderlyweb.db.AppConfig
+import org.vaccineimpact.orderlyweb.db.Config
+import org.vaccineimpact.orderlyweb.errors.MissingParameterError
+import org.vaccineimpact.orderlyweb.errors.OrderlyFileNotFoundError
+import org.vaccineimpact.orderlyweb.guessFileType
+
+class DocumentController(context: ActionContext,
+                         private val config: Config,
+                         private val files: FileSystem) : Controller(context)
+{
+    constructor(context: ActionContext) : this(context, AppConfig(), Files())
+    fun getDocument(): Boolean
+    {
+        val path = context.splat()?.joinToString("/")
+                ?: throw MissingParameterError("Path to document not provided")
+
+        val fileName = context.splat()!!.last()
+
+        val inline = context.queryParams("inline")?.toBoolean() ?: false
+
+        val absoluteFilePath = "${this.config["documents.root"]}/$path"
+
+        if (!files.fileExists(absoluteFilePath))
+        {
+            throw OrderlyFileNotFoundError(fileName)
+        }
+
+        val response = context.getSparkResponse().raw()
+
+        context.addDefaultResponseHeaders(guessFileType(fileName))
+        if (!inline)
+        {
+            context.addResponseHeader("Content-Disposition", "attachment; filename=\"$path\"")
```

Comment:
Will this behave weirdly with a path rather than just the fileName? Does it convert the path into a longer file name on download?

Reply:
This is what we do for report artefacts, it seems that firefox and chrome at least convert file separators into underscores. See e.g. artefact downloads here: https://montagu.vaccineimpact.org/reports/report/native-diagnostics-burden-report-drafts/20200207-132755-fb64cea2/#downloads

Reply:
Although arguably it's more useful for artefacts than it is here, maybe we don't need the path info here?